Xwt.XamMac/Xwt.Mac.CellViews/CompositeCell.cs

```diff
@@ -135,202 +141,158 @@ void ICopiableObject.CopyFrom (object other)
+							Messaging.void_objc_msgSend_Int64 (cell.Handle, selSetBackgroundStyle.Handle, (long)value);
+						else
+							Messaging.void_objc_msgSend_int (cell.Handle, selSetBackgroundStyle.Handle, (int)value);
+					} else
+						cell.NeedsDisplay = true;
 			}
 		}
-		
-		public override void DrawInteriorWithFrame (CGRect cellFrame, NSView inView)
-		{
-			// FIXME: although ObjectValue seems to be set and Fill called correctly,
-			//        the table flickers without an additional Fill call, especially
-			//        on expansion/collapsing with partially hidden cells (row wise).
-			//        Cocoa seems to be resetting some NSCell bits, which may be
-			//        related to the deprecated NSCell mode.
-			if (tablePosition != null)
-				Fill ();
-			CGContext ctx = NSGraphicsContext.CurrentContext.GraphicsPort;
-			ctx.SaveState ();
-			ctx.AddRect (cellFrame);
-			ctx.Clip ();
-			foreach (CellPos cp in GetCells(cellFrame))
-				cp.Cell.DrawInteriorWithFrame (cp.Frame, inView);
-			ctx.RestoreState ();
-		}
-		
-		public override void Highlight (bool flag, CGRect withFrame, NSView inView)
-		{
-			foreach (CellPos cp in GetCells(withFrame)) {
-				cp.Cell.Highlight (flag, cp.Frame, inView);
-			}
-		}
-		
-		public override NSCellHit HitTest (NSEvent forEvent, CGRect inRect, NSView ofView)
-		{
-			foreach (CellPos cp in GetCells(inRect)) {
-				var h = cp.Cell.HitTest (forEvent, cp.Frame, ofView);
-				if (h != NSCellHit.None)
-					return h;
-			}
-			return NSCellHit.None;
-		}
 
-		public override bool TrackMouse (NSEvent theEvent, CGRect cellFrame, NSView controlView, bool untilMouseUp)
-		{
-			var c = GetHitCell (theEvent, cellFrame, controlView);
-			if (c != null)
-				return c.Cell.TrackMouse (theEvent, c.Frame, controlView, untilMouseUp);
-			else
-				return base.TrackMouse (theEvent, cellFrame, controlView, untilMouseUp);
-		}
+		static readonly Selector sizeToFitSel = new Selector ("sizeToFit");
 
-		public CGRect GetCellRect (CGRect cellFrame, NSCell cell)
+		protected virtual bool SizeToFit (NSView view)
 		{
-			foreach (var c in GetCells (cellFrame)) {
-				if (c.Cell == cell)
-					return c.Frame;
-			}
-			return CGRect.Empty;
-		}
-
-		CellPos GetHitCell (NSEvent theEvent, CGRect cellFrame, NSView controlView)
-		{
-			foreach (CellPos cp in GetCells(cellFrame)) {
-				var h = cp.Cell.HitTest (theEvent, cp.Frame, controlView);
-				if (h != NSCellHit.None)
-					return cp;
+			if (view.RespondsToSelector (sizeToFitSel)) {
+				Messaging.void_objc_msgSend (view.Handle, sizeToFitSel.Handle);
+				return true;
 			}
-			return null;
+			return false;
 		}
 		
 		IEnumerable<CellPos> GetCells (CGRect cellFrame)
 		{
-			if (direction == Orientation.Horizontal) {
+			int nexpands = 0;
+			double requiredSize = 0;
+			double availableSize = cellFrame.Width;
 
-				int nexpands = 0;
-				double requiredSize = 0;
-				double availableSize = cellFrame.Width;
+			var sizes = new Dictionary<ICellRenderer, double> ();
```

Comment:
Sizes used to be a dictionary due to having to access by `ICellRenderer` later on. Can't this be a simple list? Is a cell reusable?